### PR TITLE
Feature/secure random value generation

### PR DESCRIPTION
**Pull request by BavyaMittal**
Title: Replaced `os.urandom` with `secrets.token_bytes` for secure random value generation.

Summary:
This pull request replaces all occurrences of os.urandom with the secrets package to improve the security and readability of random value generation. The secrets package is specifically designed for cryptographic purposes, making it a more suitable choice for these use cases. All required imports for secrets have been added.

Testing:
Ran pytest tests/ to verify functionality. All existing tests pass except one, which fails due to exceeding the Hypothesis deadline. The failure is unrelated to these changes and can be addressed by increasing the Hypothesis deadline from 200ms to 500ms.

Error Details for Test: 
```python
hypothesis.errors.DeadlineExceeded: Test took 253.70ms, which exceeds the deadline of 200.00ms
Falsifying example: test_key_derivation_from_password(
    password='00000000',
  salt=b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00)

### CODE REVIEW

File: .DS_Store
Comment (Veinar) [inline .DS_Store:1]: Please remove this 😃

File: envcloak/bin/Activate.ps1
Comment (Veinar) [inline envcloak/bin/Activate.ps1:1]: Please remove this 😃

File: envcloak/bin/activate
Comment (Veinar) [inline envcloak/bin/activate:1]: Please remove this 😃

File: envcloak/bin/activate.csh
Comment (Veinar) [inline envcloak/bin/activate.csh:1]: Please remove this 😃

File: envcloak/bin/activate.fish
Comment (Veinar) [inline envcloak/bin/activate.fish:1]: Please remove this 😃

File: envcloak/bin/pip
Comment (Veinar) [inline envcloak/bin/pip:1]: Please remove this 😃

File: envcloak/bin/pip3
Comment (Veinar) [inline envcloak/bin/pip3:1]: Please remove this 😃

File: envcloak/bin/pip3.11
Comment (Veinar) [inline envcloak/bin/pip3.11:1]: Please remove this 😃

File: envcloak/bin/python
Comment (Veinar) [inline envcloak/bin/python:1]: Please remove this 😃

File: envcloak/bin/python3
Comment (Veinar) [inline envcloak/bin/python3:1]: Please remove this 😃

File: envcloak/bin/python3.11
Comment (Veinar) [inline envcloak/bin/python3.11:1]: Please remove this 😃

**Review by Veinar — CHANGES_REQUESTED** [review]
Hi @BavyaMittal 👋,

Everything looks great, the test using hypothesis is not yet fully validated and may cause problems, so don't bother ☺️.

I would just ask you to remove `venv` because it got into the commit, either by accident or by my bad instructions in CONTRIBUTING.md. And thank you, for your work, many thanks!

**Comment by BavyaMittal** [comment]
Sure! I have deleted the required files.

**Review by Veinar — APPROVED** [review]
Now it is perfect 🙌

**Comment by Veinar** [comment]
Merged, and it will be used within next release!
Thank you @BavyaMittal once more 🙌

**Pull request merged**